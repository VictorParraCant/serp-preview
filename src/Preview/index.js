import React from 'react'
import {default as PT} from 'prop-types'
// import cx from 'classnames'

const Title = ({ content }) => <h3>{content}</h3>

const MetaDescription = ({ content }) => <div className="metadescription">{content}</div>

const Url = ({ content }) => <cite>{content}</cite>

const Preview = ({ searchEngine, title, metadescription, url, link }) =>
  <div id="serp-preview" className={searchEngine}>
    <a href={link ? link : '#'}>
      <Title content={title} />
      <Url content={url} />
    </a>
    <MetaDescription content={metadescription} />
  </div>

Preview.defaultProps = {
  searchEngine: 'google',
  metadescription: null,
  link: null
}

Preview.title = {
  /** Search engine */
  searchEngine: PT.oneOf(['google', 'bing', 'yahoo']),
  /** Title */
  title: PT.string.isRequired,
  /** Meta Description */
  metadescription: PT.oneOfType([
    PT.string,
    PT.arrayOf(PT.node),
    PT.node
  ]),
  /** Url */
  url: PT.string.isRequired,
  /** Link for src */
  link: PT.string
}

export default Preview
