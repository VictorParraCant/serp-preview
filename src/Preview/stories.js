import React from 'react'
import { storiesOf } from '@storybook/react'
import SerpPreview from './'

const testResults = [
  {
    title: 'ðŸ¥‡React â€“ A JavaScript library for building user interfaces',
    metadescription: 'React makes it painless to create interactive UIs. Design simple views for each state in your application, and React will efficiently update and render just the right components when your data changes. Declarative views make your code more predictable and easier to debug.',
    url: 'reactjs.org'
  },
  {
    title: 'facebook/react: A declarative, efficient, and flexible ... - GitHub',
    metadescription: 'React. Â· React is a JavaScript library for building user interfaces. Declarative: React makes it painless to create interactive UIs. Design simple views for each state in your application, and React will efficiently update and render just the right components when your data changes.',
    url: 'github.com â€º facebook â€º react'
  },
  {
    title: 'React Native Â· A framework for building native apps using React',
    metadescription: 'A framework for building native apps using React.',
    url: 'reactnative.dev'
  }
]


storiesOf('Preview')
  .add('Google', () => (
    <figure>
      { testResults.map(({ title, metadescription, url }, idx) =>
        <SerpPreview key={idx} title={title} metadescription={metadescription} url={url} />
      )}
    </figure>
  ))
  .add('Bing', () => (
    <figure>
      { testResults.map(({ title, metadescription, url }, idx) =>
        <SerpPreview key={idx} searchEngine="bing" title={title} metadescription={metadescription} url={url} />
      )}
    </figure>
  ))
  .add('Yahoo', () => (
    <figure>
      { testResults.map(({ title, metadescription, url }, idx) =>
        <SerpPreview key={idx} searchEngine="yahoo" title={title} metadescription={metadescription} url={url} />
      )}
    </figure>
  ))
