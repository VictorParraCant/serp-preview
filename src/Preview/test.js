import React from 'react'
import Preview from './'
import renderer from 'react-test-renderer'

it('show default preview', () => {
  const tree = renderer
    .create(
      <Preview
        title="test title"
        metadescription="meta description"
        url="test.com" />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('show preview for bing', () => {
  const tree = renderer
    .create(
      <Preview
        searchEngine="bing"
        title="test title"
        metadescription="meta description"
        url="test.com" />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('show preview for yahoo', () => {
  const tree = renderer
    .create(
      <Preview
        searchEngine="yahoo"
        title="test title"
        metadescription="meta description"
        url="test.com" />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
