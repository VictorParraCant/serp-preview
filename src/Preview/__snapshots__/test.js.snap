// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`show default preview 1`] = `
<div
  className="google"
  id="serp-preview"
>
  <a
    href="#"
  >
    <h3>
      test title
    </h3>
    <cite>
      test.com
    </cite>
  </a>
  <div
    className="metadescription"
  >
    meta description
  </div>
</div>
`;

exports[`show preview for bing 1`] = `
<div
  className="bing"
  id="serp-preview"
>
  <a
    href="#"
  >
    <h3>
      test title
    </h3>
    <cite>
      test.com
    </cite>
  </a>
  <div
    className="metadescription"
  >
    meta description
  </div>
</div>
`;

exports[`show preview for yahoo 1`] = `
<div
  className="yahoo"
  id="serp-preview"
>
  <a
    href="#"
  >
    <h3>
      test title
    </h3>
    <cite>
      test.com
    </cite>
  </a>
  <div
    className="metadescription"
  >
    meta description
  </div>
</div>
`;
